oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,328 @@
+discussion in the Overhang.io forums <https://discuss.overhang.io/t/migrating-from-bitnami-to-tutor/31>`__.
+
+Assuming the source installation is running the same release as the target one, it would be a reasonably straightforward
+matter of launching a new Tutor stack (with all the desired customization applied, of course), dumping the state of the
+databases, then importing them.  Nevertheless, the exact steps need to be performed, tested, and written down.
+
+
+Criterion 7. Project health
+...........................
+
+Tutor as an open source project is `3 and a half years old
+<https://github.com/overhangio/tutor/commit/bdd1a41f6207275bd24c98136fc567af2ea48b7f>`__.  It saw a peak of activity `in
+2019 <https://github.com/overhangio/tutor/graphs/code-frequency>`__, and has since then maintained a steady pace of
+improvement. The company behind it, `Overhang.IO <https://overhang.io/>`__, is declaredly committed to open source
+values and offers `Tutor deployment <https://overhang.io/tutor/>`__ as a service, and thus have a financial incentive to
+continue maintaining it.
+
+It is, however, still effectively maintained `by a single developer
+<https://github.com/overhangio/tutor/graphs/contributors>`__: Regis Behmo. This poses a significant risk: if for some
+unforeseen reason Regis can no longer maintain Tutor, the Open edX community would again be orphaned. Thus, it is
+suggested that some form of co-governance/co-maintenance be instituted. Regis has `tentatively agreed
+<https://discuss.openedx.org/t/lets-talk-about-the-native-installation/3269/5>`__.  Hopefully, with more people and
+companies using Tutor, more contributors will be willing to take on this responsibility.
+
+Tutor's codebase is technically up-to-date, well organized, and maintained. There is no detectable technical debt.
+
+
+Conclusion of the evaluation
+............................
+
+
+Positives
+`````````
+
+At the time of writing, Tutor is the only existing alternative to ``edx/configuration``. Luckily for the community, it
+ticks a lot of the boxes that qualify a good replacement:
+
+-  It can do single-server production deployments well
+-  It allows for automated deployment as well as, or better than, ``edx/configuration``
+-  It supports Koa, and new releases are added quickly
+-  It allows for theme customization, running edx-platform forks, and installing custom requirements
+-  It is extensible via plugins
+-  Actively maintained by a committed developer; codebase has no technical debt
+-  Licensed under the AGPL v3
+
+
+Omissions
+`````````
+
+There is work to be done before it can be considered a full replacement:
+
+-  There must be a documented way to run highly-available, multi-server production deployments, including a way to run
+   automated rolling upgrades with no downtime
+-  Migration from the native installation must be documented and/or automated
```

Comment:
I already anticipate that it will not be a single-click migration.

Reply:
I'd suggest to strike "and/or automated". The current native installation does not provide any automated migration between releases, so in terms of expectation management I'd say it's perfectly reasonable to expect downstream operators to put in manual work for a migration towards a new release that also happens to be deployed with Tutor.

Reply:
Agreed.  I'll remove "and/or automated".